src/client/mod.rs

```diff
@@ -372,8 +416,8 @@ impl OAuth1 for Signer {
     fn signing_key(&self) -> String {
         format!(
             "{}&{}",
-            urlencoding::encode(&self.consumer_secret),
-            urlencoding::encode(&self.secret)
+            urlencoding::encode(self.consumer_secret.as_ref()),
+            urlencoding::encode(self.secret.as_ref())
```

Comment:
What is the difference between the two version? To me, it is easier to read with the previous version

Reply:
With `String` with have `ops::Deref<Target = str>` but with the proposed change we are generic on `T: AsRef<str>` which is a broader